### DISCUSSION TITLE
Return string value directly, or reduce list to string if value is a list

### DISCUSSION THREAD

**Discussion by manuelgrabowski** [mikefarah/yq#1592] [Q&A] [answered]
I'm struggling to get something rather simple to work, I'm afraid. My input is this file: https://gitlab.com/gitlab-com/www-gitlab-com/-/raw/master/data/stages.yml

I'm using the following to find out the PM for a certain group, for example for "Geo":

```
curl -s 'https://gitlab.com/gitlab-com/www-gitlab-com/-/raw/master/data/stages.yml' | yq '.stages.[].groups.[] | select(.name | test("(?i)'"geo"'")) | .name + ": " +  .pm'
```

That works fine. However, in the file you'll see that not every `pm:` entry is a string. If you search for `pipeline_execution`, for example, you'll see that it's put as a list. Replacing "geo" with "execution" in the one liner above will expectedly result in an error: `Error: !!seq (stages.verify.groups.pipeline_execution.pm) cannot be added to a !!str ()`

I can come up with a version that handles this and returns this list as a comma separated string, but then it stops working for the ones that give a string directly. I can't figure out how to write something that handles both cases.

**Comment by mikefarah** [comment, marked as answer]
Easiest way I can think of solving that is to add the `.pm` to an empty array. If its a string, you'll have an array with a single string. If it's an array of strings, then it will simple be an array concatenation, and you end up with the array of strings. From there, you can join the string if desired:

```
./yq '.stages[].groups[] | select(.name | test("(?i)'"exe"'")) | {.name: [] + .pm}'
```

yields:
```
"Pipeline Execution":
  - Jackie Porter
```

with join:

```
./yq '.stages[].groups[] | select(.name | test("(?i)'"exe"'")) | {.name: [] + .pm | join(", ")}' 
```

yields:
```
"Pipeline Execution": Jackie Porter
```

**Reply by manuelgrabowski** [reply]
That is absolutely perfect, thank you very much. I even had the idea that this would be a way to solve it on an abstract level, but due to lack of experience I didn't realize that `[] + .item` was all it takes to actually do it.
